feat: support multiple file extentions

KhudaDad414 opened the pull request:
**Description**
- support `.yaml`, `.json` and `.yml` file extensions for the AsyncAPI file.
- add core configuration to the config file.

fixes #23

KhudaDad414 commented:
@Souvikns need your feedback here. 😁

fmvilas commented on src/lib/configs.ts:
```diff
@@ -50,6 +50,26 @@ async function loadConfigsFromFile() {
+  }
+  
+  if(indexOfFoundFile === -1){
+    throw new Error('AsyncAPI file was not found at the root of your project. you can set its path in config file.')
```
Would be great to provide help on how "you can set its path in config file". Have a look at https://wix-ux.com/when-life-gives-you-lemons-write-better-error-messages-46c5223e1a2f.

Also, when this error throws, it will be showing the stack trace and all the stuff that comes with throwing an error. This information is not relevant as it's not a failure in the code. Please consider "catching" the error in a function above and simply show a beautiful and informative error message using https://github.com/asyncapi/glee/blob/master/src/lib/logger.ts#L123.

fmvilas commented on src/lib/configs.ts:
```diff
@@ -50,6 +50,26 @@ async function loadConfigsFromFile() {
+  if(indexOfFoundFile === -1){
+    throw new Error('AsyncAPI file was not found at the root of your project. you can set its path in config file.')
+  }
+  return path.resolve(baseDir, files[indexOfFoundFile])
```
This is a way to avoid the "Function Call Object Injection Sink" warning:

```suggestion
  return path.resolve(baseDir, files.at(indexOfFoundFile))
```

fmvilas reviewed (changes requested):
You've got additional warnings and I added a few comments on how you can solve them.

KhudaDad414 commented:
@fmvilas thanks for the review. 🙇  I noticed some configurations weren't documented. I have added them and also changed the format so it matches the config file's overall format. anything you want to add?

fmvilas commented on src/lib/configs.ts:
```diff
@@ -38,16 +44,30 @@ async function loadConfigsFromFile() {
 
+export function findSpecFile(baseDir: string): string{
+  const files = ['asyncapi.yaml', 'asyncapi.json', 'asyncapi.yml']
+  const foundFiles = files.filter(file => existsSync(path.resolve(baseDir, file)))
```
You should not be using `existsSync` because:

1. It only tests the existence of the path but it doesn't test if you have read permission. So it may happen that the file is there but you can't open it.
2. It tests the existence of the **path**. A path might be a file or a directory. We want to make sure it's a file.

Therefore I recommend using [fs.accessSync](https://nodejs.org/api/fs.html#fsaccesssyncpath-mode) and [fs.statSync](https://nodejs.org/api/fs.html#fsstatsyncpath-options) (which allows you to check for [isFile()](https://nodejs.org/api/fs.html#statsisfile)).

fmvilas reviewed (changes requested):
I found another problem. Other than that, seems ready 👍

fmvilas commented on src/lib/configs.ts:
```diff
@@ -13,41 +14,68 @@ let GLEE_FUNCTIONS_DIR: string
  */
 async function loadConfigsFromFile() {
-  if (!existsSync(GLEE_CONFIG_FILE_PATH)) return 
+  if (!isFileReadable(GLEE_CONFIG_FILE_PATH)) return 
```
Good catch 💯

fmvilas commented on src/lib/configs.ts:
```diff
@@ -13,41 +14,68 @@ let GLEE_FUNCTIONS_DIR: string
+  }  else if(foundFiles.length > 1) {
+    errorMessage = "Multiple AsyncAPI files found. Please choose one in you config file (https://github.com/asyncapi/glee/blob/master/docs/config-file.md)."
+  } else {
+    errorMessage = "Unable fo find the AsyncAPI file. Please make sure it's in your project's directory or set its path in the config file(https://github.com/asyncapi/glee/blob/master/docs/config-file.md)."
```
```suggestion
    errorMessage = "Unable fo find the AsyncAPI file. Please make sure it's in your project's directory or set its path in the config file (https://github.com/asyncapi/glee/blob/master/docs/config-file.md)."
```

fmvilas reviewed (changes requested):
Last minor change. It's done already 🚀

fmvilas commented:
/rtm

fmvilas reviewed (approved):
Forgot to approve 🤦

fmvilas commented:
/rtm
